src/entry/state/entryActions.ts

```diff
@@ -51,8 +52,27 @@ export const fetchEntry = (accession: string) => async (dispatch: Dispatch) => {
   fetchData(url)
     .then(({ data }: { data: UniProtkbAPIModel }) => {
       dispatch(receiveEntry(accession, data));
-    }) /* eslint-disable no-console */
-    .catch(error => console.error(error));
+    })
+    .catch((error) => {
+      const { status } = error.response;
+
+      switch (status) {
+        case 400:
+        case 404:
+          history.push('/page-not-found');
```

Comment:
and here you can re-use the page name as set in the enum (see `App.tsx` comment)

Reply:
I tried to do that but `LocationToPath.get()` which is internally a `Map.get()` does not guarantee a return value, therefore `history.push()` is rejecting it.

Any suggestions?

Reply:
I see the issue with using a `Map`. I'm going to commit onto your branch with a rework of the urls.

Reply:
I have just pushed a commit to resolve this.